packages/api/src/command/medical/tcm-encounter/get-tcm-encounters.ts

```diff
@@ -1,88 +1,109 @@
 import { TcmEncounterModel } from "../../../models/medical/tcm-encounter";
-import { Pagination } from "../../pagination";
-import { buildDayjs } from "@metriport/shared/common/date";
+import { Pagination, sortForPagination } from "../../pagination";
+import { omit } from "lodash";
 
 /**
  * Add a default filter date far in the past to guarantee hitting the compound index
  */
 const DEFAULT_FILTER_DATE = new Date("2020-01-01T00:00:00.000Z");
 
-export type GetTcmEncountersCmd = {
-  cxId: string;
-  after?: string;
-  pagination: Pagination;
-};
-
-export type CountTcmEncountersCmd = {
-  cxId: string;
-  after?: string;
-};
+export interface TcmEncounterQueryData extends TcmEncounterModel {
+  dataValues: TcmEncounterModel["dataValues"] & {
+    patient_data: PatientModel["dataValues"]["data"];
+  };
+}
 
-type TcmEncounterForDisplay = TcmEncounter & {
-  patientName: string;
+export type TcmEncounterResult = TcmEncounterModel["dataValues"] & {
+  patientData: PatientModel["dataValues"]["data"];
 };
 
 export async function getTcmEncounters({
   cxId,
   after,
   pagination,
-}: GetTcmEncountersCmd): Promise<TcmEncounterForDisplay[]> {
-  const where: WhereOptions<TcmEncounterModel> = {
-    cxId,
-    admitTime: {
-      [Op.gt]: after ? buildDayjs(after).toDate() : DEFAULT_FILTER_DATE,
-    },
-  };
+}: {
+  cxId: string;
+  after?: string;
+  pagination: Pagination;
+}): Promise<TcmEncounterResult[]> {
+  const tcmEncounterTable = TcmEncounterModel.tableName;
+  const patientTable = PatientModel.tableName;
 
-  const rows = await TcmEncounterModel.findAll({
-    where,
-    include: [
-      {
-        model: PatientModel,
-        as: "PatientModel",
-        attributes: ["id", "cxId", "data"],
-      },
-    ],
-    limit: pagination.count + 1, // Get one extra to determine if there's a next page
-    order: [["admitTime", "DESC"]],
-  });
+  const sequelize = TcmEncounterModel.sequelize;
+  if (!sequelize) throw new Error("Sequelize not found");
 
-  const items = rows.map((row: TcmEncounterModel) => {
-    const patient = (row.get("PatientModel") as PatientModel).dataValues.data;
+  const { toItem, fromItem } = pagination;
 
-    /** Hack to get around Sequelize type inference not seeing associations */
-    const encounterData = { ...row.dataValues, PatientModel: undefined };
-    delete encounterData.PatientModel;
+  /**
+   * ⚠️ Always change this query and the count query together.
+   */
+  const queryString = `
+      SELECT tcm_encounter.*, patient.data as patient_data
+      FROM ${tcmEncounterTable} tcm_encounter
+      INNER JOIN ${patientTable} patient 
+      ON tcm_encounter.patient_id = patient.id
+      WHERE tcm_encounter.cx_id = :cxId
+      AND tcm_encounter.admit_time > :afterDate
+      ${toItem ? ` AND tcm_encounter.id >= :toItem` : ""}
+      ${fromItem ? ` AND tcm_encounter.id <= :fromItem` : ""}
+      ORDER BY tcm_encounter.id ${toItem ? "ASC" : "DESC"}
```

Comment:
Can we call `getPaginationSorting()`? It's easier to explain in that function why we do it the way we do it, instead of wondering why are we ordering differently based on the filter (I was doing that, even had a comment ready here until I looked at the existing pagination and remembered).